src/renderer/gl_engine/tvgGlShaderSrc.cpp

```diff
@@ -207,14 +207,107 @@ std::string STR_RADIAL_GRADIENT_VARIABLES = TVG_COMPOSE_SHADER(
 
 std::string STR_RADIAL_GRADIENT_MAIN = TVG_COMPOSE_SHADER(
     out vec4 FragColor;                                                                                     \n
+                                                                                                            \n
+    mat3 radial_matrix(vec2 p0, vec2 p1) {                                                                  \n
+        return mat3(0.0, -1.0, 0.0, 1.0, 0.0, 0.0, 0.0, 0.0, 1.0) *                                         \n
+         inverse(mat3(p1.y - p0.y, p0.x - p1.x, 0.0, p1.x - p0.x, p1.y - p0.y,                              \n
+                      0.0, p0.x, p0.y, 1.0));                                                               \n
+    }                                                                                                       \n
```

Comment:
can you please provide any refs, how did you get this transformation?
also,, are you sure, that we should compose this matrix every time for each pixel or we can compose it once in CPU side?

Reply:
The matrix comes from [skia conical gradient](https://github.com/google/skia/blob/main/src/shaders/gradients/SkConicalGradient.cpp#L48). 
It can be done in CPU, but requires some other floating point precision judgment code I think it's too troublesome so I put it in Shader :(